aviary/core/aviary_group.py

```diff
@@ -982,6 +983,49 @@ def add_post_mission_systems(self, verbosity=None):
             promotes_outputs=[('mass_resid', Mission.Constraints.MASS_RESIDUAL)],
         )
 
+        ecomp = om.ExecComp(
+            'excess_fuel_capacity = total_fuel_capacity - unusable_fuel - overall_fuel',
+            total_fuel_capacity={'units': 'lbm'},
+            unusable_fuel={'units': 'lbm'},
+            overall_fuel={'units': 'lbm'},
+            excess_fuel_capacity={'units': 'lbm'},
+        )
+
+        post_mission.add_subsystem(
+            'excess_fuel_constraint',
+            ecomp,
+            promotes_inputs=[
+                ('total_fuel_capacity', Aircraft.Fuel.TOTAL_CAPACITY),
+                ('unusable_fuel', Aircraft.Fuel.UNUSABLE_FUEL_MASS),
+                ('overall_fuel', Mission.Summary.TOTAL_FUEL_MASS),
+            ],
+            promotes_outputs=[('excess_fuel_capacity', Mission.Constraints.EXCESS_FUEL_CAPACITY)],
+        )
+
+        # determine if the user wants the excess_fuel_capacity constraint active and if so add it to the problem
+        try:
+            # for backwards compatability check to see if variable exists, if not assume default value = False
+            ignore_capacity_constraint = self.aviary_inputs.get_val(
+                Aircraft.Fuel.IGNORE_FUEL_CAPACITY_CONSTRAINT, units='unitless'
+            )
+        except:
+            warnings.warn(
```

Comment:
Probably want to move these warnings under the Verbosity level check so that they don't warn when quiet.